_posts/news/2025-01-07-announcing-scalajs-1.18.0.md

```diff
@@ -0,0 +1,139 @@
+These are not externally visible, except for users who directly manipulate the IR and/or linker.
+They pave the way for upcoming changes that are still ongoing work.
+
+Read on for more details.
+
+<!--more-->
+
+## Getting started
+
+If you are new to Scala.js, head over to [the tutorial]({{ BASE_PATH }}/tutorial/).
+
+If you need help with anything related to Scala.js, you may find our community [in `#scala-js` on Discord](https://discord.com/invite/scala) and [on Stack Overflow](https://stackoverflow.com/questions/tagged/scala.js).
+
+Bug reports can be filed [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Release notes
+
+If upgrading from Scala.js 0.6.x, make sure to read [the release notes of Scala.js 1.0.0]({{ BASE_PATH }}/news/2020/02/25/announcing-scalajs-1.0.0/) first, as they contain a host of important information, including breaking changes.
+
+This is a **minor** release:
+
+* It is backward binary compatible with all earlier versions in the 1.x series: libraries compiled with 1.0.x through 1.17.x can be used with 1.18.0 without change.
+* It is *not* forward binary compatible with 1.17.x: libraries compiled with 1.18.0 cannot be used with 1.17.x or earlier.
+* It is *not* entirely backward source compatible: it is not guaranteed that a codebase will compile *as is* when upgrading from 1.17.x (in particular in the presence of `-Xfatal-warnings`).
+
+As a reminder, libraries compiled with 0.6.x cannot be used with Scala.js 1.x; they must be republished with 1.x first.
+
+## Changes with compatibility concerns
+
+### Enforce stricter rules on the use of `this` in the IR of constructors
+
+Exceptionally, **this is theoretically a binary-breaking change**.
+It is possible for previously valid IR (Intermediate Representation, as published in binaries on Maven Central) to be rejected by Scala.js 1.18.0.
+
+We now enforce, during IR checking (by default, only enabled in `fullLink` builds), that constructors abide by stricter rules.
+Before calling their `super` constructor (or delegate `this()` constructor), they may not refer to the `this` instance.
+The only exception is to *assign* a field, as in `this.foo = bar`.
+This restriction is in line with a similar rule in JVM bytecode.
+
+Unfortunately, our compiler plugin for Scala 2.x was previously generating IR that is invalid according to the new rule in one corner case situation.
+If there is a `try..catch` expression inside an argument to the `super` constructor, the compiler may lift it to a separate method.
+We compiled that method as an *instance* method, using the `this` value, instead of compiling it as a *static* method.
+
+Here is an example:
+
+{% highlight scala %}
+import scala.util.control.Breaks._
+
+class Parent(val i: Int) {
+  def this(p: Parent) = this(p.i)
+  if (i > 40) throw new IllegalArgumentException
+}
+class Child extends Parent({
+  val p = try new Parent(42) catch { case _: IllegalArgumentException => break() }
```

Comment:
IMO this is very confusing, because the `Parent` class is also used as a value. How about:

```suggestion
  val p = try Integer.parseInt("foo") catch { case _: NumberFormatException => break() }
```

And remove the check in `Parent`?